step-templates/aws-linux-install-tentacle.json

```diff
@@ -3,14 +3,14 @@
ssion.cidr,Values='0.0.0.0/0' | convertfrom-json\n    \n    if($sgCheck.SecurityGroups.count -eq 0)\n    {\n\t\tWrite-Host \"Creating SG Rule\"\n    \taws ec2 authorize-security-group-ingress --group-name $sgName --ip-permissions IpProtocol=tcp,ToPort=10933,FromPort=10933,IpRanges='[{CidrIp=0.0.0.0/0,Description=\"OctopusListeningTentacle\"}]'\n\t}\n    else\n    {\n    \tWrite-Host \"Found Existing SG Rule\"\n    }\n}\n\nWrite-Verbose \"hostname: $hostname`nnoListen: $noListen\"\n\n$remoteScript = @\"\n{\n\t\"commands\": [\n    \t\"#!/bin/bash\",\n\t\t\"curl -L https://octopus.com/downloads/latest/Linux_x64TarGz/OctopusTentacle -o /tmp/tentacle-linux_x64.tar.gz -fsS\",\n\t\t\"if [ ! -d \\\"/opt/octopus\\\" ]; then sudo mkdir /opt/octopus; fi\",\n\t\t\"tar xvzf /tmp/tentacle-linux_x64.tar.gz -C /opt/octopus\",\n\t\t\"rm /tmp/tentacle-linux_x64.tar.gz\",\n\t\t\"cd /opt/octopus/tentacle\",\n\t\t\"sudo /opt/octopus/tentacle/Tentacle create-instance --config '/etc/octopus/default/tentacle-default.config'\",\n\t\t\"sudo chmod a+rwx /etc/octopus/default/tentacle-default.config\",\n\t\t\"/opt/octopus/tentacle/Tentacle new-certificate --if-blank\",\n\t\t\"/opt/octopus/tentacle/Tentacle configure --port 10933 --noListen False --reset-trust --app '/home/Octopus/Applications'\",\n\t\t\"/opt/octopus/tentacle/Tentacle configure --trust $octoThumb\",\n\t\t\"echo 'Registering the Tentacle $name with server $serverUri in environment $environments with role $roles'\",\n\t\t\"/opt/octopus/tentacle/Tentacle register-with --server '$serverUri' --apiKey '$apikey' $environments $roles --comms-style $comStyle --name '$tentacleName' --force\",\n\t\t\"sudo /opt/octopus/tentacle/Tentacle service --install --start\"\n\t]\n}\n\"@\n\nWrite-Host \"Installing tentacle on remote machine\"\n$guid = (new-guid).guid\nSet-Content -Value $remoteScript -Path \"$env:Temp/$guid.json\"\n\nwrite-verbose $remoteScript\n\nwrite-verbose \"aws ssm send-command --document-name `\"AWS-RunShellScript`\" --instance-ids `\"$instanceId`\" --parameters `\"file://$env:Temp/$guid.json`\"\"\ntry {\n\t$cmdResponse = aws ssm send-command --document-name \"AWS-RunShellScript\" --instance-ids \"$instanceId\" --parameters \"file://$env:Temp/$guid.json\" --query \"Command\" --output json | convertfrom-json\n    $cmdId = $cmdResponse.CommandId\n    $errorResponse = aws ssm get-command-invocation --command-id \"$cmdId\" --instance-id \"$instanceId\" --output json | convertfrom-json\n    \n    while($errorResponse.Status -eq 'InProgress')\n    {\n    \twrite-verbose \"`nStatus: $($errorResponse.Status)\"\n    \t$errorResponse = aws ssm get-command-invocation --command-id \"$cmdId\" --instance-id \"$instanceId\" --output json | convertfrom-json\n    }\n    \n    write-verbose \"`nErrorResponse: $errorResponse`n\"\n    \n    if(![string]::IsNullOrEmpty($errorResponse.StandardErrorContent))\n    {\n    \tthrow $errorResponse.StandardErrorContent\n    }\n}\nfinally {\n\tremove-item \"$env:Temp\\$guid.json\"\n}\n",
```

Comment:
The `$portNumber` isn't referenced in the script body. Instead `10933` is still referenced:

```powershell
$noListen = "--port 10933 --noListen `"false`""
```

```powershell
$sgCheck = aws ec2 describe-security-groups --group-names $sgName --output json --filters Name=ip-permission.from-port,Values=10933 Name=ip-permission.cidr,Values='0.0.0.0/0' | convertfrom-json
```

```powershell
aws ec2 authorize-security-group-ingress --group-name $sgName --ip-permissions IpProtocol=tcp,ToPort=10933,FromPort=10933,IpRanges='[{CidrIp=0.0.0.0/0,Description="OctopusListeningTentacle"}]'```